packages/teleport-generator-core/src/validator/index.ts

```diff
@@ -38,6 +42,66 @@ export default class Validator {
 
     return { valid: true, errorMsg: '' }
   }
+
+  public validateComponentContent(input: any): ValidationResult {
+    const verifyProps = checkContent(input, 'prop')
+    const verifyState = checkContent(input, 'state')
+
+    checkForDuplicates(input)
```

Comment:
same as above about the naming. once we will test more than just props and state here, duplicates will not be suggestive enough